synthesizer/program/src/resources/credits.aleo

```diff
@@ -216,82 +236,85 @@ finalize bond_public:
     /* Committee */
 
     // Check if the caller is a validator.
-    contains committee[r0] into r19;
+    contains committee[r0] into r21;
     // Enforce the caller is *not* a validator.
-    assert.eq r19 false;
+    assert.eq r21 false;
 
     // Get the stake for the specified validator.
     // If the validator does not exist, this finalize scope will fail.
-    get committee[r1] into r20;
+    get committee[r1] into r22;
     // Ensure that the validator is open to stakers.
-    assert.eq r20.is_open true;
+    assert.eq r22.is_open true;
 
     // Increment the stake for the specified validator.
-    add r20.microcredits r2 into r21;
+    add r22.microcredits r3 into r23;
     // Construct the updated committee state.
-    cast r21 r20.is_open into r22 as committee_state;
+    cast r23 r22.is_open into r24 as committee_state;
 
     // Check if the delegator is already bonded to the validator.
-    contains bonded[r0] into r23;
+    contains bonded[r0] into r25;
     // If the delegator is already bonded to the validator, jump to the `continue_bond_delegator` logic.
-    branch.eq r23 true to continue_bond_delegator;
+    branch.eq r25 true to continue_bond_delegator;
+
     // Get the number of delegators.
-    get.or_use metadata[aleo1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqanmpl0] 0u32 into r24;
+    get.or_use metadata[aleo1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqanmpl0] 0u32 into r26;
     // Increment the number of bonded delegators by one.
-    add r24 1u32 into r25;
+    add r26 1u32 into r27;
     // Determine if the number of delegators is less than or equal to 100_000.
-    lte r25 100_000u32 into r26;
+    lte r27 100_000u32 into r28;
     // Enforce that the number of delegators is less than or equal to 100_000.
-    assert.eq r26 true;
+    assert.eq r28 true;
     // Set the new number of delegators.
-    set r25 into metadata[aleo1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqanmpl0];
+    set r27 into metadata[aleo1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqanmpl0];
+    // Set the withdrawal address.
+    set r2 into withdraw[r0];
```

Comment:
There is 2 instances of `set _ into withdraw[_]` because it denotes the validator case and the delegator case.

This should only be called once during the initial bond of the validator or delegator. This is the same as the metadata mappings only being set during the initial bond.